chia/data_layer/data_store.py

```diff
@@ -449,12 +521,14 @@ async def insert(
         *,
         lock: bool = True,
         status: Status = Status.PENDING,
+        optimized: bool = False,
     ) -> bytes32:
         async with self.db_wrapper.locked_transaction(lock=lock):
             was_empty = await self.table_is_empty(tree_id=tree_id, lock=False)
             root = await self.get_tree_root(tree_id=tree_id, lock=False)
 
-            if not was_empty:
+            # If `optimized`, skip this check.
+            if not was_empty and not optimized:
                 # TODO: is there any way the db can enforce this?
                 pairs = await self.get_pairs(tree_id=tree_id, lock=False)
                 if any(key == node.key for node in pairs):
```

Comment:
it seems to me like there should be a sql index for the key, and we should just ensure it's unique. That seems a lot cheaper than `get_pairs()`. If we have that, do we still need a way to skip the key uniqueness check?

Reply:
there can be multiple identical `keys` in the table - for different `tree_id`. so just finding the key isn't enough - we need to also check `tree_id`. anyway, I think this falls out of scope of this PR - I needed to make changes to insert only so I can test for a big tree. I'll change `optimized` to `skip_expansive_checks`.